src/util/irep_ids.h

```diff
@@ -35,8 +35,29 @@ Author: Reuben Thomas, reuben.thomas@me.com
 
 #ifdef USE_DSTRING
 
-#define IREP_ID_ONE(the_id) extern const dstringt ID_##the_id;
-#define IREP_ID_TWO(the_id, str) extern const dstringt ID_##the_id;
+enum class idt:unsigned
+{
+#define IREP_ID_ONE(the_id) id_##the_id,
+#define IREP_ID_TWO(the_id, str) id_##the_id,
+
+#include "irep_ids.def" // NOLINT(build/include)
+};
+
+#ifdef __GNUC__
```

Comment:
Clang, MSVC 2015+ also have this. Suggest stealing the testing logic used at https://github.com/diffblue/cbmc/blob/develop/src/nonstd/optional.hpp#L87